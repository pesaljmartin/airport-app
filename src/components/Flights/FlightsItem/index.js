import React from "react";
import moment from "moment";

import "./index.scss";

const FlightsItem = props => {
  const { departure, origin, destination, detail } = props.flight;
  return (
    <div className="flights-item-wrapper">
      <div>
        <div className="flights-item-wrapper__route">
          <p>Destination: {origin && origin.value}</p>
          <p className="arrow-wrapper">
            <i className="arrow up" />
          </p>
          <p>{destination && destination.value}</p>
        </div>
        <p>
          Departure:
          {departure && moment(departure).format("DD MMM YYYY hh:mm a")}
        </p>
      </div>
      <div>
        <p>Flight number: {detail && detail.flight_number}</p>
        <p>
          Boarding time:
          {detail && moment(detail.boarding).format("DD MMM YYYY hh:mm a")}
        </p>
      </div>
      <div>
        <p>Gate: {detail && detail.gate}</p>
        <p>Seat: {detail && detail.seat}</p>
      </div>
    </div>
  );
};

export default FlightsItem;
