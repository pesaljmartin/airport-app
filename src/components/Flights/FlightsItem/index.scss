.flights-item-wrapper {
  padding: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 5px;
  display: flex;

  > div:not(:first-child) {
    border-left: 2px solid #000000;
    padding: 0 10px;
  }

  > div:first-child {
    padding-right: 10px;
  }

  > div:not(:last-child) {
    width: 40%;
  }

  &__route {
    display: flex;
    align-items: center;

    .arrow-wrapper {
      padding: 0 8px 0 5px;
    }
  }
}
