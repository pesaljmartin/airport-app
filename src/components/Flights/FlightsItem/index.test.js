import React from "react";
import FlightsItem from "./index";
import renderer from "react-test-renderer";

test("FlightsItem comoponent snapshot", () => {
  const component = renderer.create(<FlightsItem flight={{}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
