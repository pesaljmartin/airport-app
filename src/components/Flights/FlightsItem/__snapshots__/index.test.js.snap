// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightsItem comoponent snapshot 1`] = `
<div
  className="flights-item-wrapper"
>
  <div>
    <div
      className="flights-item-wrapper__route"
    >
      <p>
        Destination: 
      </p>
      <p
        className="arrow-wrapper"
      >
        <i
          className="arrow up"
        />
      </p>
      <p />
    </div>
    <p>
      Departure:
    </p>
  </div>
  <div>
    <p>
      Flight number: 
    </p>
    <p>
      Boarding time:
    </p>
  </div>
  <div>
    <p>
      Gate: 
    </p>
    <p>
      Seat: 
    </p>
  </div>
</div>
`;
